pkg/webhook/mutatingwebhook.go

```diff
@@ -0,0 +1,118 @@
+/*
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package webhook
+
+import (
+	"context"
+	"encoding/json"
+	"github.com/fluid-cloudnative/fluid/pkg/ddc/base"
+	"github.com/fluid-cloudnative/fluid/pkg/utils/kubeclient"
+	"github.com/fluid-cloudnative/fluid/pkg/webhook/plugins"
+	"net/http"
+	ctrl "sigs.k8s.io/controller-runtime"
+
+	corev1 "k8s.io/api/core/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/webhook/admission"
+)
+
+// +kubebuilder:webhook:path=/mutate-v1-pod,mutating=true,failurePolicy=ignore,groups="",resources=pods,verbs=create,versions=v1,name=fluid-pod-admission-webhook
+
+// MutatingHandler mutates a pod and has implemented admission.DecoderInjector
+type MutatingHandler struct {
+	Client client.Client
+	// A decoder will be automatically injected
+	decoder *admission.Decoder
+}
+
+func NewMutatingHandler(c client.Client) *MutatingHandler {
+	return &MutatingHandler{
+		Client: c,
+	}
+}
+
+// Handle is the mutating logic of pod
+func (a *MutatingHandler) Handle(ctx context.Context, req admission.Request) admission.Response {
+	var setupLog = ctrl.Log.WithName("handle")
+	pod := &corev1.Pod{}
+	err := a.decoder.Decode(req, pod)
+	if err != nil {
+		setupLog.Error(err, "unable to decoder pod from req")
+		return admission.Errored(http.StatusBadRequest, err)
+	}
+
+	// check whether should inject
+	if pod.Labels["Fluid-Injection"] == "disabled" {
```

Comment:
labels k/v and injection determination should be defined in one place
e.g. : pkg/common/labels.go
```go
const (
    LabelKeyPrefix = "fluid.io"
)
const (
  LabelKeyFluidInejction = LabelKeyPrefix + "/fluid-injection"
)

const(
  FlagDisabled = "disabled"
  FlagEnable = "enabled"
)

func TargetHit(l map[string]string, target string) bool {
	if an == nil {
		return false
	}
	if v, ok := an[target]; ok {
		v := strings.ToLower(v)
		if v == FlagEnable {
			return true
		}
	}
	return false
}
func TargetHitWithValue(an map[string]string, target, value string) bool {
	if an == nil {
		return false
	}
	if v, ok := an[target]; ok {
		return v == value
	}
	return false
}
```
```
if common.TargetHitWithValue(pod.Labels,common.LabelKeyFluidInejction,common.FlagDisabled){}
```

Reply:
@allenhaozi maybe we can enhance this in next PR.